apps/nextra/pages/en/build/smart-contracts/book/structs-and-resources.mdx

```diff
@@ -236,15 +236,19 @@ module 0x2::m {
 }
 ```
 
-If the field is implicitly copyable, the dot operator can be used to read fields of a struct without
-any borrowing. (Only scalar values with the `copy` ability are implicitly copyable.)
+The dot operator can be used to read and copy any copyable field of a struct without explicit
+borrowing and dereferencing:
 
 ```move
 module 0x2::m {
   fun example() {
     let foo = Foo { x: 3, y: true };
     let x = foo.x;  // x == 3
     let y = foo.y;  // y == true
+
+    let bar = Bar { foo };
+    let foo2: Foo = *&bar.foo; // `Foo` must be copyable
+    let foo3: Foo = bar.foo;   // same as the statement above
   }
 }
```

Comment:
We may want to put this earlier, I think it's important to show people the easy mode copy first rather than the `*&` paradigm that was existing in early move.

Do we have a section that talks about non-copyable fields, would do a `move`, and therefore you can't do that?  (but you can reference them)

Reply:
Could you clarify "we may want to put this earlier"? Did you mean swap the statements defining `foo2` and `foo3`, or something else?

> Do we have a section that talks about non-copyable fields, would do a move, and therefore you can't do that? (but you can reference them)

The closest I could find was this: https://aptos.dev/en/build/smart-contracts/book/references#reading-and-writing-through-references

Reply:
@gregnazario would you be able to follow up with the above?

Reply:
I defer, and let's just continue with what's here.